### PR TITLE
igraph: random improvements

### LINKED ISSUE #166573: Feedback regarding igraph package
This is not a bug report, merely some hopefully useful feedback regarding how igraph is packaged in Nix:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/igraph/default.nix

It seems that you are simply grabbing a certain commit from GitHub. The recommended way is to use the release tarballs instead (unless you are working with a dev version). For example, for 0.9.7, the release tarball is here:

https://github.com/igraph/igraph/releases/download/0.9.7/igraph-0.9.7.tar.gz

This is not exactly the same as the snapshot of the v0.9.7 tag at https://github.com/igraph/igraph/archive/refs/tags/0.9.7.tar.gz

Specifically, the release tarball:

 - Contains pre-built documentation, making it unnecessary to build the documentation locally (it's slow), and eliminating multiple build dependencies (`fop`, `libxslt`, `python3`, `sourceHighlight`, `xmlto`).
 - Contains the parser sources, eliminating the need for `bison` and `flex` as well.
 - Comes "ready to build", making the hack on this line unnecessary: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/igraph/default.nix#L38 Normally, igraph is expected to be built either from an actual git repo (not merely a snapshot of its contents) or the release tarball.

Of course, if you are already aware of this, and are building from a specific commit on purpose, feel free to keep it that way. However, please do make sure that the version written to `IGRAPH_VERSION` is correct. For example, if you set it to a later commit that doesn't match any release tag, this should be indicated in the version by adding a part following a `-` sign. E.g., the version generated for given commits now looks like this: `0.10.0-dev+aba150fb`.

[This comment](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/igraph/default.nix#L33) is outdated as the build system is no longer autotools and there is no `tools/getversion.sh`. The `IGRAPH_VERSION` file is read with CMake.

After [this line](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/igraph/default.nix#L77) you may want to add `IGRAPH_USE_INTERNAL_PLFIT=ON`. Since igraph 0.9.6, it is possible to use an external plfit, however, Nix does not currently contain this package.

You may want to make an explicit decision about whether to build with OpenMP and set `IGRAPH_OPENMP_SUPPORT` accordingly. The default is `AUTO` as the clang that comes with Xcode does not support OpenMP.

CC @MostAwesomeDude @dotlambda

### PR DESCRIPTION
**Pull request by dotlambda**
###### Description of changes
* remove outdated comment
* use plfit from Nixpkgs
* unconditionally enable OpenMP

closes https://github.com/NixOS/nixpkgs/issues/166573

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @szhorvat

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -6374,6 +6374,10 @@ in {
 
   plexwebsocket = callPackage ../development/python-modules/plexwebsocket { };
 
+  plfit = toPythonModule (pkgs.plfit.override {
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:6377]: I'm not sure this is the correct name cause plfit is a different package on PyPI: https://pypi.org/project/plfit/

Comment (szhorvat) [reply pkgs/top-level/python-packages.nix:6377]: Just to clarify, what igraph depends on is this package: https://github.com/ntamas/plfit

Comment (dotlambda) [reply pkgs/top-level/python-packages.nix:6377]: > Just to clarify, what igraph depends on is this package: https://github.com/ntamas/plfit

That's the one I packaged.
But there's a Python package of the same name, which could lead to confusion.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 167058` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>hal-hardware-analyzer</li>
    <li>python39Packages.scikit-tda</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>igraph</li>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.igraph</li>
    <li>python310Packages.kmapper</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-files</li>
    <li>python39Packages.explorerscript</li>
    <li>python39Packages.igraph</li>
    <li>python39Packages.kmapper</li>
    <li>python39Packages.skytemple-dtef</li>
    <li>python39Packages.skytemple-files</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>plfit</li>
    <li>python310Packages.plfit</li>
    <li>python39Packages.plfit</li>
  </ul>
</details>

**Review by MostAwesomeDude — APPROVED** [review]
LGTM

**Comment by dotlambda** [comment]
> 11 packages failed to build

not igraph's fault

**Pull request merged**